File: backend/src/NLC.Application/DTOs/AuthDtos.cs
namespace NLC.Application.DTOs;

public record LoginRequest(string Email, string Password);

public record LoginResponse(
    string AccessToken,
    string RefreshToken,
    DateTime ExpiresAt,
    UserDto User
);

public record RefreshRequest(string RefreshToken);

public record UserDto(
    Guid   Id,
    string Email,
    string Role,
    Guid[] AssignedWarehouseIds
);

File: backend/src/main/java/com/nlc/web/dto/AuthDtos.java
package com.nlc.web.dto;

import com.nlc.domain.enums.Enums.UserRole;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import java.time.OffsetDateTime;
import java.util.UUID;

public final class AuthDtos {
    private AuthDtos() {}

    public record LoginRequest(
        @NotBlank @Email String email,
        @NotBlank String password
    ) {}

    public record RefreshRequest(@NotBlank String refreshToken) {}

    public record LoginResponse(
        String accessToken,
        String refreshToken,
        OffsetDateTime expiresAt,
        UserDto user
    ) {}

    public record UserDto(
        UUID id,
        String email,
        String name,
        String role,
        String[] assignedWarehouseIds
    ) {}
}
